File: src/Lessons/1-Iterations/binaryGap.go
package main

import (
	"fmt"
	"strconv"
)

func main() {
	fmt.Println(solution(1202))
}

func solution(someNumber int) int {
	numberBinary := strconv.FormatInt(int64(someNumber), 2) //convert to binary
	fmt.Println(numberBinary)
	gapSize := 0
	tempGapSize := 0

	for index := 0; index < len(numberBinary); index++ {
		if string(numberBinary[index]) == "1" {
			if tempGapSize > gapSize {
				gapSize = tempGapSize
			}
			tempGapSize = 0
		} else {
			tempGapSize++
		}
	}
	return gapSize
}

File: src/Lessons/1-Iterations/binaryGap.js
//Binary Gap
//Time complexity 0^N
function solution(someNumber) {
	var numberBinary = someNumber.toString(2); //Convert to a binary format string

	var gapSize = 0; //Store our current largest gap
	var tempGapSize = 0; //Temporary store our working gapsize
	//Start our loop
	for (let index = 0; index < numberBinary.length; index++) {
		const bit = numBinary[index];
		if (bit == '1') {
			if (tempGapSize > gapSize) {
				//if our tempgapsize is bigger than our gapsize update
				gapSize = tempGapSize;
			}
			// Reset to zero
			tempGapSize = 0;
		} else {
			//Add another our size with one as the bit is another zero
			tempGapSize++;
		}
	}
	return gapSize;
}